### PR TITLE
fix: dispose of resources in time

### PR DESCRIPTION
**Pull request by Xazin**
This PR aims to dispose of resources when possible, and fix some miscelleanous issues.

Closes: #2611 

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database/application/layout/layout_listener.dart
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/application/layout/layout_listener.dart:1]: Removed as it was unused

File: frontend/appflowy_flutter/lib/plugins/database/application/setting/setting_controller.dart
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/application/setting/setting_controller.dart:1]: Removed as it was unused

File: frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/filter/filter_menu.dart [hunk 1]
```diff
@@ -35,7 +35,12 @@ class FilterMenu extends StatelessWidget {
           final List<Widget> children = [];
           children.addAll(
             state.filters
-                .map((filterInfo) => FilterMenuItem(filterInfo: filterInfo))
+                .map(
+                  (filterInfo) => FilterMenuItem(
+                    key: ValueKey(filterInfo.filter.id),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/filter/filter_menu.dart:40]: Resolves issue where deleting filters of same name and field type only works once, then the rest are stuck

File: frontend/appflowy_flutter/lib/plugins/database/board/presentation/board_page.dart [hunk 2]
```diff
@@ -58,6 +58,9 @@ class BoardPageTabBarBuilderImpl implements DatabaseTabBarItemBuilder {
   ) =>
       const SizedBox.shrink();
 
+  @override
+  void dispose() {}
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database/board/presentation/board_page.dart:62]: is it okay to define the dispose function with an empty body in its superclass? so its subclass doesn't need to override it.

**Review by LucasXu0 — APPROVED** [review]

**Pull request merged**